server/executor/trigger/grpc_carrier.go

```diff
@@ -0,0 +1,54 @@
+package trigger
+
+import (
+	"github.com/kubeshop/tracetest/server/model"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+type GRPCHeaderCarrier struct {
```

Comment:
I would remove this entire implementation and just use https://pkg.go.dev/go.opentelemetry.io/otel/propagation#MapCarrier